--- /dev/null
+++ b/configs/InvizBox_2_defconfig
@@ -0,0 +1,13 @@
+CONFIG_ARM=y
+CONFIG_ARCH_SUNXI=y
+CONFIG_MACH_SUN8I_H3=y
+CONFIG_DRAM_CLK=672
+CONFIG_DRAM_ZQ=3881979
+CONFIG_DRAM_ODT_EN=y
+CONFIG_MACPWR="PD6"
+CONFIG_MMC_SUNXI_SLOT_EXTRA=2
+CONFIG_DEFAULT_DEVICE_TREE="/sun8i-h3-invizbox2"
+CONFIG_SPL=y
+CONFIG_SUN8I_EMAC=y
+CONFIG_USB_EHCI_HCD=y
+CONFIG_SYS_USB_EVENT_POLL_VIA_INT_QUEUE=y
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -331,7 +331,8 @@
 	sun8i-h3-nanopi-m1.dtb \
 	sun8i-h3-nanopi-m1-plus.dtb \
 	sun8i-h3-nanopi-neo.dtb \
-	sun8i-h3-nanopi-neo-air.dtb
+	sun8i-h3-nanopi-neo-air.dtb \
+	sun8i-h3-invizbox2.dtb
 dtb-$(CONFIG_MACH_SUN8I_R40) += \
 	sun8i-r40-bananapi-m2-ultra.dtb
 dtb-$(CONFIG_MACH_SUN8I_V3S) += \
diff --git a/arch/arm/dts/sun8i-h3-invizbox2.dts b/arch/arm/dts/sun8i-h3-invizbox2.dts
new file mode 100644
index 0000000000..2c1a7b8142
--- /dev/null
+++ b/arch/arm/dts/sun8i-h3-invizbox2.dts

@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: (GPL-2.0+ or MIT)
+
+/*
+ * Copyright (C) 2018 Invizbox
+ */
+/dts-v1/;
+#include "sun8i-h3.dtsi"
+#include "sunxi-common-regulators.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/pinctrl/sun4i-a10.h>
+
+/ {
+       model = "Invizbox 2";
+       compatible = "invizbox,invizbox-2", "allwinner,sun8i-h3";
+
+       aliases {
+               serial0 = &uart0;
+               ethernet0 = &emac;
+       };
+
+       chosen {
+               stdout-path = "serial0:115200n8";
+       };
+
+       leds {
+               compatible = "gpio-leds";
+
+               vpn {
+                       label = "invizbox:green:vpn";
+                       gpios = <&pio 0 20 GPIO_ACTIVE_HIGH>;
+               };
+
+               internet {
+                       label = "invizbox:green:internet";
+                       gpios = <&pio 0 21 GPIO_ACTIVE_HIGH>;
+                       default-state = "on";
+               };
+
+               wifi {
+                       label = "invizbox:green:wifi";
+                       gpios = <&r_pio 0 10 GPIO_ACTIVE_HIGH>;
+               };
+       };
+
+       reg_vcc_wifi: reg_vcc_wifi {
+               compatible = "regulator-fixed";
+               regulator-min-microvolt = <3300000>;
+               regulator-max-microvolt = <3300000>;
+               regulator-name = "vcc-wifi";
+               enable-active-high;
+               gpio = <&r_pio 0 7 GPIO_ACTIVE_HIGH>;
+       };
+
+       rtl8376rb {
+               compatible = "realtek,rtl8367b";
+               gpio-sda = <&pio 3 17 GPIO_ACTIVE_HIGH>;
+               gpio-sck = <&pio 3 16 GPIO_ACTIVE_HIGH>;
+               realtek,extif1 = <1 0 1 1 1 1 1 1 2>;
+               cpu_port = <6>;
+       };
+
+       wifi_pwrseq: wifi_pwrseq {
+               compatible = "mmc-pwrseq-simple";
+               reset-gpios = <&pio 0 10 GPIO_ACTIVE_LOW>;
+               post-power-on-delay-ms = <200>;
+       };
+};
+
+&emac {
+       pinctrl-names = "default";
+       pinctrl-0 = <&emac_rgmii_pins>;
+       phy-mode = "rgmii";
+       allwinner,force-external-phy;
+       status = "okay";
+       allwinner,rx-delay-ps = <200>;
+       allwinner,tx-delay-ps = <400>;
+
+       /delete-node/ mdio;
+       /delete-node/ mdio-mux;
+
+       fixed-link {
+               speed = <1000>;
+               full-duplex;
+               pause;
+       };
+};
+
+/* Redefine the pins to free MDIO pins for RTL switch driver */
+&emac_rgmii_pins {
+       pins = "PD0", "PD1", "PD2", "PD3", "PD4",
+               "PD5", "PD7", "PD8", "PD9", "PD10",
+               "PD12", "PD13", "PD15";
+};
+
+&mmc1 {
+       pinctrl-names = "default";
+       pinctrl-0 = <&mmc1_pins_a>;
+       vmmc-supply = <&reg_vcc_wifi>;
+       mmc-pwrseq = <&wifi_pwrseq>;
+       bus-width = <4>;
+       non-removable;
+       status = "okay";
+
+       brcmf: brcmf@1 {
+               reg = <1>;
+               compatible = "brcm,brcmfmac";
+               interrupt-parent = <&pio>;
+               interrupts = <0 11 IRQ_TYPE_LEVEL_LOW>; /* PA11 */
+               interrupt-names = "host-wake";
+       };
+};
+
+&mmc2 {
+       pinctrl-names = "default";
+       pinctrl-0 = <&mmc2_8bit_pins>;
+       vmmc-supply = <&reg_vcc3v3>;
+       vqmmc-supply = <&reg_vcc3v3>;
+       bus-width = <8>;
+       non-removable;
+       status = "okay";
+};
+
+&ohci1 {
+       status = "okay";
+};
+
+&ohci2 {
+       status = "okay";
+};
+
+&ehci1 {
+       status = "okay";
+};
+
+&ehci2 {
+       status = "okay";
+};
+
+&uart0 {
+       pinctrl-0 = <&uart0_pins_a>;
+       status = "okay";
+};
+
+&usbphy {
+       status = "okay";
+};
